spark-job/src/main/java/org/apache/cassandra/diff/DiffJob.java

```diff
@@ -87,15 +89,28 @@ public void run(JobConfiguration configuration, JavaSparkContext sc) {
         ClusterProvider targetProvider = ClusterProvider.getProvider(configuration.clusterConfig("target"), "target");
         String sourcePartitioner;
         String targetPartitioner;
+        List<KeyspaceTablePair> tablesToCompare = configuration.filteredKeyspaceTables();
         try (Cluster sourceCluster = sourceProvider.getCluster();
              Cluster targetCluster = targetProvider.getCluster()) {
             sourcePartitioner = sourceCluster.getMetadata().getPartitioner();
             targetPartitioner = targetCluster.getMetadata().getPartitioner();
+
+            if (!sourcePartitioner.equals(targetPartitioner)) {
+                throw new IllegalStateException(String.format("Cluster partitioners do not match; Source: %s, Target: %s,",
+                                                              sourcePartitioner, targetPartitioner));
+            }
+
+            if (configuration.shouldAutoDiscoverTables()) {
+                Schema sourceSchema = new Schema(sourceCluster.getMetadata(), configuration);
+                Schema targetSchema = new Schema(targetCluster.getMetadata(), configuration);
+                Schema commonSchema = sourceSchema.intersect(targetSchema);
+                if (commonSchema.size() != sourceSchema.size()) {
+                    logger.warn("Found tables that only exist in either source or target cluster. Ignoring those tables for comparision. ");
```

Comment:
Nit: I wonder if this needs to be a warning, rather than just info.

Reply:
The diff job expects the schema from the compared clusters to be the same. If we found schema mismatch, the comparison might not be valid. Therefore, I choose to log it at `warn` level to highlight the unexpected.

Reply:
Well, it used to expect the two schema to be the same, now it expects them to intersect :) It's not a big deal, I was just thinking that it might not be that unusual for some cases - e.g. if you run the metadata db on either the source or target clusters.

Reply:
Yes. It could be sort of annoying when metadata db is one of the clusters. It is easy to reason about IMO. 
Maybe add the difference of 2 sets in the log statement?